spark/src/main/java/org/apache/iceberg/spark/data/vectorized/IcebergArrowColumnVector.java

```diff
@@ -0,0 +1,753 @@
+              default:
+                throw new UnsupportedOperationException(
+                    "Unsupported base type for decimal: " + primitive.getPrimitiveTypeName());
+            }
+          default:
+            throw new UnsupportedOperationException(
+                "Unsupported logical type: " + primitive.getOriginalType());
+        }
+      } else {
+        switch (primitive.getPrimitiveTypeName()) {
+          case FIXED_LEN_BYTE_ARRAY:
+          case BINARY:
+            return new DictionaryBinaryAccessor((IntVector) vector);
+          case INT32:
+            return new DictionaryIntAccessor((IntVector) vector);
+          case FLOAT:
+            return new DictionaryFloatAccessor((IntVector) vector);
+          case INT64:
+            return new DictionaryLongAccessor((IntVector) vector);
+          case DOUBLE:
+            return new DictionaryDoubleAccessor((IntVector) vector);
+          default:
+            throw new UnsupportedOperationException("Unsupported type: " + primitive);
+        }
+      }
+    } else {
+      if (vector instanceof BitVector) {
+        return new BooleanAccessor((BitVector) vector);
+      } else if (vector instanceof TinyIntVector) {
+        return new ByteAccessor((TinyIntVector) vector);
+      } else if (vector instanceof SmallIntVector) {
+        return new ShortAccessor((SmallIntVector) vector);
+      } else if (vector instanceof IntVector) {
+        return new IntAccessor((IntVector) vector);
+      } else if (vector instanceof BigIntVector) {
+        return new LongAccessor((BigIntVector) vector);
+      } else if (vector instanceof Float4Vector) {
+        return new FloatAccessor((Float4Vector) vector);
+      } else if (vector instanceof Float8Vector) {
+        return new DoubleAccessor((Float8Vector) vector);
+      } else if (vector instanceof IcebergArrowVectors.DecimalArrowVector) {
+        return new DecimalAccessor((IcebergArrowVectors.DecimalArrowVector) vector);
+      } else if (vector instanceof IcebergArrowVectors.VarcharArrowVector) {
+        return new StringAccessor((IcebergArrowVectors.VarcharArrowVector) vector);
+      } else if (vector instanceof IcebergArrowVectors.VarBinaryArrowVector) {
+        return new BinaryAccessor((IcebergArrowVectors.VarBinaryArrowVector) vector);
+      } else if (vector instanceof DateDayVector) {
+        return new DateAccessor((DateDayVector) vector);
+      } else if (vector instanceof TimeStampMicroTZVector) {
+        return new TimestampAccessor((TimeStampMicroTZVector) vector);
+      } else if (vector instanceof ListVector) {
+        ListVector listVector = (ListVector) vector;
+        return new ArrayAccessor(listVector);
+      } else if (vector instanceof StructVector) {
+        StructVector structVector = (StructVector) vector;
+        ArrowVectorAccessor structAccessor = new StructAccessor(structVector);
+        childColumns = new ArrowColumnVector[structVector.size()];
```

Comment:
I think this should be moved out of this method. There's no need for this method to have side-effects, and this makes it hard to understand how this class works because `childColumns` appears to be uninitialized when reading the constructor. I'd much rather add `childColumns()` to the `ArrowVectorAccessor` interface and initialize like this:

```
  this.accessor = VectorAccessors.get(vector);
  this.childColumns = accessor.childColumns();
```

You could also remove the `childColumns` field and always call `accessor.childColumns()`.